go.mod

```diff
@@ -18,58 +17,55 @@ require (
 	k8s.io/klog v1.0.0
-	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
+	k8s.io/kube-openapi v0.0.0-20200805222855-6aeccd4b50c6
 	knative.dev/networking v0.0.0-20200922180040-a71b40c69b15
 	knative.dev/pkg v0.0.0-20200922164940-4bf40ad82aab
 	knative.dev/serving v0.18.0
 	sigs.k8s.io/controller-runtime v0.6.3
 )
 
 replace (
-	k8s.io/api => k8s.io/api v0.18.8
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.18.8
-	k8s.io/apimachinery => k8s.io/apimachinery v0.18.8
-	k8s.io/apiserver => k8s.io/apiserver v0.18.8
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.18.8
-	k8s.io/client-go => k8s.io/client-go v0.18.8
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.18.8
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.18.8
-	k8s.io/code-generator => k8s.io/code-generator v0.18.8
-	k8s.io/component-base => k8s.io/component-base v0.18.8
-	k8s.io/cri-api => k8s.io/cri-api v0.18.8
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.18.8
+	k8s.io/api => k8s.io/api v0.19.2
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.19.2
+	k8s.io/apimachinery => k8s.io/apimachinery v0.19.2
+	k8s.io/apiserver => k8s.io/apiserver v0.19.2
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.19.2
+	k8s.io/client-go => k8s.io/client-go v0.19.2
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.19.2
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.19.2
+	k8s.io/code-generator => k8s.io/code-generator v0.19.2
+	k8s.io/component-base => k8s.io/component-base v0.19.2
+	k8s.io/cri-api => k8s.io/cri-api v0.19.2
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.19.2
 	k8s.io/klog => k8s.io/klog v1.0.0
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.18.8
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.18.8
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.18.8
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.18.8
-	k8s.io/kubectl => k8s.io/kubectl v0.18.8
-	k8s.io/kubelet => k8s.io/kubelet v0.18.8
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.18.8
-	k8s.io/metrics => k8s.io/metrics v0.18.8
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.18.8
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.19.2
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.19.2
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.19.2
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.19.2
+	k8s.io/kubectl => k8s.io/kubectl v0.19.2
+	k8s.io/kubelet => k8s.io/kubelet v0.19.2
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.19.2
+	k8s.io/metrics => k8s.io/metrics v0.19.2
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.19.2
 	k8s.io/test-infra => k8s.io/test-infra v0.0.0-20200803112140-d8aa4e063646
 	k8s.io/utils => k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
 	// based on https://github.com/openshift/cluster-ingress-operator/pull/429/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R34
```

Comment:
Can you help remove the following `controller-runtime` pin and bump up to 0.7.0? I think we no longer need it once move to 1.19.2.

Reply:
I gave it a try but it turns out we use a bunch of deprecated packages from `controller-runtime` that are deleted in `0.7.0`. These are `pkg/runtime/log` and `pkg/runtime/signals`. The issue mentioned in this comment (https://github.com/kubernetes-sigs/controller-runtime/issues/1033) is fixed only from 0.7.0 onwards. I created https://github.com/kubeflow/kfserving/issues/1307 to track the migration away from the deprecated packages and the update of this dependency.